src/components/PopupWithForm.js

```diff
@@ -1,33 +1,41 @@
-import { Popup } from "./Popup.js";
-
-export default class PopupWithForm extends Popup {
+import Popup from "./Popup";
+class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
-    super(popupSelector);
-    this._form = this._popupElement.querySelector(".modal__form");
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".popup__form");
     this._handleFormSubmit = handleFormSubmit;
+    this._modalSaveButton = this._popupElement.querySelector(".popup__button");
+    this._saveButtonText = this._modalSaveButton.textContent;
   }
 
-  _getInputValues() {
-    const values = {};
-    this._form.querySelectorAll(".modal__input").forEach((input) => {
-      values[input.name] = input.value;
+  _getInputValue() {
+    const inputs = this._popupForm.querySelectorAll(".popup__input");
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.